pkg/remote/client_test.go

```diff
@@ -172,6 +175,40 @@ func TestNewWriteRequestBody(t *testing.T) {
 	assert.Contains(t, string(b), `label1`)
 }
 
+func TestNewWriteRequestBodyWithStaleMarker(t *testing.T) {
+	t.Parallel()
+
+	ts := []*prompb.TimeSeries{
+		{
+			Labels: []*prompb.Label{{Name: "label1", Value: "val1"}},
+			Samples: []*prompb.Sample{{
+				Value:     math.Float64frombits(0x7ff0000000000002),
```

Comment:
To avoid redefining this, maybe export the `remotewrite.staleNaN` var, or if an import cycle is a problem, put it in a common package?